src/shared/reboot-util.c

```diff
@@ -107,3 +125,63 @@ int shall_restore_state(void) {
 
         return r > 0 ? ret : true;
 }
+
+static int xen_kexec_loaded(void) {
+#if HAVE_XENCTRL
+        size_t size;
+        _cleanup_close_ int privcmd_fd = -1;
+        _cleanup_close_ int buf_fd = -1;
+        void *buffer;
+        xen_kexec_status_t *status;
+        int r;
+
```

Comment:
given this will run on every kexec, we'll always get at least one spurious debug log message - maybe it's worth to do a `if if (access("/proc/xen", F_OK) < 0) return -EOPNOTSUPP` as the very first thing? Without logging the result. That way there's no extra noise in the "normal" case?